students/KennanY/Lesson6/MailRoomPt4.py

```diff
@@ -0,0 +1,137 @@
+        return
+
+def func_Send_Thankyou():
+    ''' Send a Thank you note'''
+    print('Thank you')
+    value=0
+    name = input('Type donor name or ''list'' to see all the names=')
+    if name == 'list':
+        # List all donors if user enters 'list'
+        for donor in people.keys():
+            print(donor)
+    else:  # It is assumed that name entered in this case
+        # Prompt for donation amount
+        try:
+            value = float(input('Please enter the donation amount='))
+            # Is it in the db?
+            # If in database, append value else create a new record
+            if name in people:
+                #Add the donation
+                getlist = people[name]
+                getlist.append(value)
+        except:
+            print("Invalid value entered!")
+            return
+        else:
+            #Add the name and the donation
+            people[name] = value
+        # Print the email now
+        PrintEMail(name, value)
+
+def func_Create_Report():
+    """Print donation history"""
+    donations= []
+    for donors in people:
+        entry=[]
+        entry.append(donors)
+        for item in people[donors]:
+            entry.append(item)
+
+        donations.append(entry)
+
+    #Sort the donations
+    report_data = [[person[0], sum(person[1:]), len(person[1:]), sum(person[1:]) / len(person[1:])]
+                  for person in donations]
+    sorted_data = sorted(report_data, key=lambda x: x[1], reverse=True)
+
+    #Print the header
+    print('{0:25}{1:15}{2:15}{3:15}'.format('Donor Name', '| Total Gifts', '| Num Gifts', '| Average Gift'))
+    print('=' * 50)
+
+    for donor in donations:
+        total_amount=sum(donor[1:])
+        total_donation=len(donor)-1
+        avg_donation=total_amount/total_donation
+        print('{:<25s}  ${:>18,.2f}  {:>15d}  ${:>18,.2f}'.format( donor[0], total_amount, total_donation,
+                                                                      avg_donation))
+
+def  func_Send_Letters():
+    print('Write the letters to file in ' + os.getcwd())
+    Generate_Letters(people)
+
+def func_Quit():
+    global flag
+    flag=False
+    print('Quiting ... DONE!')
+
+def print_donors():
+    print('Print Donors')
+
+
+#Display the menu
+def showmenu ():
+    """Displays the menu options"""
+    while (flag):
+        print('\n')
+        print('1 - Send a Thank You')
+        print('2 - Create a Report')
+        print('3 - Send letters to everyone')
+        print ('4 -  Quit')
+        response=int(input("Please select and option (1,2, 3 or 4)"))
+        # Dict implementation of the menu
+        switch_func_dict = {
+            1: func_Send_Thankyou,
+            2: func_Create_Report,
+            3: func_Send_Letters,
+            4: func_Quit,
+        }
+        if (response >= 0) and (response < 5):
+            switch_func_dict[response]()
+
+if __name__ == '__main__':
+    showmenu()
```

Comment:
@yilmazkennan - this code looks good overall, but I'm getting the same error as the previous version when adding a new donor: 'TypeError: 'float' object is not iterable'